public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -332,6 +332,7 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
         .setTitle('')
         .setOption('showTypeIcons', true)
         .setOption('showHeader', true)
+        .setData(this.state.$data?.clone())
```

Comment:
since VizPanelManager.createFor moves the data to provider to VizPanelManger level, then table should use that data. I think that this is not an ideal solution because we are already cloning the data from the sourcePanel but we have to clone here again.